### ISSUE TITLE
Own handwriting

### ISSUE THREAD

**Issue by Gerrit02** [makeyourownneuralnetwork/makeyourownneuralnetwork#37]
Hello,
even converting a picture from let's say 37K converting to a 28x28 pixel give's to much noise to let recognize the neural network the numbers correctly.

As last test did I not tested my own handwriting (because of too much noise when converting from lets say a 100x100 pixel number to a 28x28 pixel but I wrote directly in paint some numbers in 28x28 pixel format and tried to let recognize the neural network those numbers.

Eventhough here the success rate is zero too :-(
But can you tell why those numbers (see attach) are not recognized this time?

Second question, how do you succeed to convert from let's say a 37K picture to a 28x28 picture without too many noise?

[
![28_01](https://user-images.githubusercontent.com/34437030/36693525-d0333e96-1b3b-11e8-8b12-95d27bb43306.JPG)
![28_02](https://user-images.githubusercontent.com/34437030/36693526-d058bb6c-1b3b-11e8-9de8-600517169ceb.JPG)

](url)

**Comment by makeyourownneuralnetwork** [comment]
what is 37K?

**Comment by Gerrit02** [comment]
If you take a picture with a modern smartphone the size is arround 1.2 meg or even 3 meg. Then I convert the picture but is usual automatically done when you want to send the picture via mail with a smaller size (eg: 37k). So my picture is first already converted from 1.2 meg size to 37K (kilobits). Then resize to 28x28, but even then to much size.

So it would be great known how to take pictures in 28x28 format, or converting with not that many  noise.
Anyway that aside... 

Can you have a look at above pictures? Drawn in Paint, immediatly in 28x28 format, still not one number is recognized by the network... any reason for that? (network even trained with the full dataset)!

**Comment by makeyourownneuralnetwork** [comment]
My method is to draw a picture with a brush (or scan a newspaper with a number) and square crop the number so that it is in the centre of the square picture.

Then scale the image down to 28x28.

That works for me .. it doesn't when my images are not centred, or have a style of writing  not seen in the training data then it fails. This is expected as the neural network only learns what the training data shows it.

**Comment by helge42** [comment]
@gerrit: You should go through your code. The networks answer is allways 5. I think the mistake is not in the images .

**Comment by makeyourownneuralnetwork** [comment]
thanks - I will go through the code

**Comment by Gerrit02** [comment]
@helge42 it is gerrit02 (add gerrit pointed to somebody else)!

Ok when I rerun my code it seems the networks answer is still the one as what his first answer was for the first label.
So if the first answer was eg: 7 then the answers for my other labels were also 7... if it was 5 then all answers are 5.

BUT: I don't know if something is wrong with the code. I used the code from the book and ihub. And when I use the testset eg: mnist_test and mnist_test_10.csv then I have the outcome what is been predicted in the book.

I will attach some files like, eg my testset I use, the numbers itselfs where as the testset consists of and finally my code.

My testset is: MyTestSet_02.csv (csv is not a supported filetype, so I added an docx to the end, just remove that docx to have my csv file)!
The numbers the testset consists of are: een.png, twee.png and drie.png
And finally my code. You can play with yourself now as you want. Idem here, I can not upload the file, so added docs add the end, just remove the docx.

[MyTestSet_02.csv.docx](https://github.com/makeyourownneuralnetwork/makeyourownneuralnetwork/files/1778458/MyTestSet_02.csv.docx)


![drie](https://user-images.githubusercontent.com/34437030/36945253-761556c8-1fab-11e8-9560-29a0101ec1cc.png)
![een](https://user-images.githubusercontent.com/34437030/36945254-7635e028-1fab-11e8-959d-ec75dd0702fb.png)
![twee](https://user-images.githubusercontent.com/34437030/36945255-76515196-1fab-11e8-8e88-3d6b57f24c4e.png)


@Rashid below my code to debug but as said I don't think there is anything wrong with it.

[MyNeuralNetwork.ipynb.docx](https://github.com/makeyourownneuralnetwork/makeyourownneuralnetwork/files/1778459/MyNeuralNetwork.ipynb.docx)

**Comment by makeyourownneuralnetwork** [comment]
let us know if you find a problem with the code.
thanks for all your work and input.

**Comment by Gerrit02** [comment]
@helge42
in attach, you can see that I also printed the output from all output nodes. The networks answer is in this case always 1, but you can see per label the output is slightly different from the previous result. So there are really some calculations going on.
And as said when I used the test set from the book, I have the expected outcome. Nothing changes within the code.

![allresults](https://user-images.githubusercontent.com/34437030/36996434-24d0b098-20b7-11e8-9f03-31bccbc62851.JPG)

@makeyourownneuralnetwork 
I added my three labels I created myself also into the training_100 dataset.
So I thought when I executed the code this time with those 3 labels (as test set) my performance should be 100% because as explained in your book, you can not use labels put in a test set that are also put in the training set. The label that was in the training set is recognized when it counters that label in the test set. So therefor the networks answer should allways match?

But nothing less is true? My outcome is the same when I train my network with the training dataset_100 with or without my 3 selfmade labels included?
Can you explain?

**Comment by makeyourownneuralnetwork** [comment]
you don't mean "label" you mean "test data record"?

the network is learning an "average" over many examples so it is entirely possible that it doesn't correctly recognise an example from the training data

**Comment by Gerrit02** [comment]
@makeyourownneuralnetwork 
Can it help then to change my epoch into eg: 1000 or more so the average is better and better after each cycle? So my train data record (included my self made records) are trained all over again. For now increased my epcoh till 200 but still same outcome?

**Comment by makeyourownneuralnetwork** [comment]
At the end of the book we discuss why havign too many epochs is not helpful - because you "overfit" the training data and can't generalise to other test data.
